packages/data-stores/src/onboard/types.ts

```diff
@@ -0,0 +1,20 @@
+import type { RequestCartProduct } from '@automattic/shopping-cart';
+
+export type DomainSuggestion = {
+	is_free: boolean;
+	domain_name: string;
+	isRecommended: boolean;
+	isBestAlternative: boolean;
+	is_premium?: boolean;
+	product_slug: string;
+};
+
+export type DomainForm = {
+	lastQuery?: string;
+	subdomainSearchResults?: DomainSuggestion[] | null;
+	loadingResults?: boolean;
+	searchResults?: DomainSuggestion[] | null;
+	hideInitialQuery?: boolean;
+};
+
+export type CartItem = Partial< RequestCartProduct > & Pick< RequestCartProduct, 'product_slug' >;
```

Comment:
If you already are importing `RequestCartProduct` here, could you just use [MinimalRequestCartProduct](https://github.com/Automattic/wp-calypso/blob/3e6c473d71c9877b5f3173c1b608bb9ffc96e368/packages/shopping-cart/src/types.ts#L219-L220) instead of recreating it as `CartItem`? It doesn't seem worth duplicating the type.